### ISSUE TITLE
Training Results

### ISSUE THREAD

**Issue by Huage001** [FoundationVision/LlamaGen#39]
Dear authors,

Thanks for your excellent work in autoregressive image generation!

I try to reproduce the training of GPT-B-256 following the instruction provided [here](https://github.com/FoundationVision/LlamaGen/blob/main/GETTING_STARTED.md). The specific command I used is:

```
torchrun \
--nnodes=1 --nproc_per_node=8 --node_rank=0 \
--master_addr=127.0.0.1 --master_port=26667 \
train_c2i.py --cloud-save-path ckpt/GPT_B --code-path dataset/imagenet_code_c2i_flip_256_ten_crop/ --image-size 256 --gpt-model GPT-B 
```

However, after training ~150 epoch on ImageNet1k, it seems that the generated results are still meaningless:

<img width="1013" alt="Screenshot 2024-07-04 at 11 20 04" src="https://github.com/FoundationVision/LlamaGen/assets/44341570/e5dbc18f-c631-4cde-805a-0464970798d1">

My environment is 8xA5000 GPUs, which is different from yours (8xA100). I wonder whether the results are sensitive to such a difference, and whether the problem would be alleviated after full training (300 epochs).

Thanks for your help in advance :)

**Comment by PeizeSun** [comment]
Hi~
This is not normal. In general, you can see meaningful images after 10 epochs training.

Can you first try to check whether the image code is correct? You can use this script [here](https://github.com/FoundationVision/LlamaGen/blob/main/tools/check_image_codes.py)

**Comment by Huage001** [comment]
Dear Peize,

Thanks for your very prompt reply! I have checked the image code using the script. Yes, it seems that the image code is normal. Here is the reconstruction results of '0.npy' and the results seem reasonable:

<img width="1017" alt="Screenshot 2024-07-04 at 17 22 04" src="https://github.com/FoundationVision/LlamaGen/assets/44341570/1f64521b-c80c-4538-8195-3270bd9e9fbe">

In addition, I can also produce reasonable images using your released checkpoints.

**Comment by PeizeSun** [comment]
Did you change the batch size, learning rate or other training parameters ?

**Comment by Huage001** [comment]
Dear Peize,

I didn't change anything in .py files. I assumed that the command for 256 resolution is the same as that for 384 and used the script above.
